src/edu/mills/cs180a/RomanNumeral.java

```diff
@@ -14,154 +14,175 @@
+
+        // throw an error if any of the chars do not exist in the map
+        for (int i = 0; i < letters.length; i++) {
+            if (!LETTERS_TO_VALUES.containsKey(String.valueOf(letters[i]))) {
+                throw new IllegalArgumentException(letters[i] + "is not a valid roman numeral");
+            }
+        }
+
+        int numericValue = LETTERS_TO_VALUES.get(String.valueOf(letters[0]));
+
+        // set the limit to one before the full length of the char array so that we don't run into
+        // any issues with variable next
+        for (int i = 0; i < letters.length - 1; i++) {
+            String current = String.valueOf(letters[i]);
+            String next = String.valueOf(letters[i + 1]);
+
+            if (LETTERS_TO_VALUES.get(current) >= LETTERS_TO_VALUES.get(next)) {
+                numericValue += LETTERS_TO_VALUES.get(next);
+            } else {
+                // Check if current starts a subtractive form (e.g., IX), where the second symbol
+                // has a higher value than the first. Subtract current twice: once because it is the
+                // start of a subtractive form and once because it was added in a previous
+                // iteration.
+                numericValue += LETTERS_TO_VALUES.get(next) - (2 * LETTERS_TO_VALUES.get(current));
+            }
+        }
+        return numericValue;
+    }// end convertFromString
+
+
     /**
-     * Returns the Roman Numeral representation of the given number.
+     * Returns the Roman Numeral representation of any given number by first determining which keys
+     * in {@link #VALUES_TO_LETTERS} the number falls between and setting the lesser key as a
+     * {@link #lowerBound}. The number, {@link #lowerBound} and a StringBuilder are passed to a
+     * helper method, which will append the appropriate Roman symbol to the StringBuilder and return
+     * the number, of which the value has decreased based on the numeric value of the appended
+     * symbol.
      *
      * @param n the number to convert
      * @return the Roman Numeral representation
-     * @throws IllegalArgumentException if the number cannot be represented
-     * by a Roman Numeral in the given range
+     * @throws IllegalArgumentException if the argument is out of bounds
      */
     @VisibleForTesting
-    protected static String convertToText(int n) 
-    {
-    	//GOAL: convert number into strings(Roman Numerals)
-    	//Refer to note in convertToInteger
-    	
-    	//acquire input, compare it to the values in the map
-    	//return the key
-    	if(n==MIN_VALUE)
-    	{
-    		LETTERS_TO_VALUES.containsValue(n);
-    	}
-    	
-    	
- 	 	return null;
-    }
-}//end RomanNumeral class
+    protected static String convertFromInt(int n) {
+        // lowerBound should be set to the Roman Numeral that closely matches the outermost place
+        // value of any given number. Ex n = 56, lowerBound = L; n = 383, lowerBound = C
+        String lowerBound = "";
```

Comment:
I see no benefit to initializing `lowerBound`, which is always set in the loop, I believe.

Reply:
I get an error if I don't have the variable initialized, is there a way for me to get around this?

Reply:
It's frustrating when students tell me they get an error without telling me what the error it is. In the future, please be more specific.

In any case, it looks like you rewrote the code.